File: termux-shared/src/main/java/com/termux/shared/shell/command/environment/ShellEnvironmentVariable.java
package com.termux.shared.shell.command.environment;

/**
 * @param name    The name for environment variable
 * @param value   The value for environment variable
 * @param escaped If environment variable {@link #value} is already escaped.
 */
public record ShellEnvironmentVariable(String name, String value,
                                       boolean escaped) implements Comparable<ShellEnvironmentVariable> {

    @Override
    public int compareTo(ShellEnvironmentVariable other) {
        return this.name.compareTo(other.name);
    }
}

File: termux-shared/src/main/java/com/termux/shared/shell/command/environment/ShellEnvironmentVariable.kt
package com.termux.shared.shell.command.environment

/**
 * @param name    The name for environment variable
 * @param value   The value for environment variable
 * @param escaped If environment variable [.value] is already escaped.
 */
@JvmRecord
data class ShellEnvironmentVariable(
    @JvmField val name: String, @JvmField val value: String,
    @JvmField val escaped: Boolean
) : Comparable<ShellEnvironmentVariable> {
    override fun compareTo(other: ShellEnvironmentVariable): Int {
        return name.compareTo(other.name)
    }
}
